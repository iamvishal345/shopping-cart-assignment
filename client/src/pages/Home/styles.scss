.home-container {
  margin-bottom: 1rem;
}
