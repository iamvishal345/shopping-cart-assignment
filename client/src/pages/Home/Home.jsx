import React, { useEffect, useState } from "react";
import "./styles.scss";
import { Card } from "components/Atoms/Card/Card";
import { Carousel } from "components/Atoms/Carousel/Carousel";
import { Head } from "Utils/Head";

const Home = () => {
  const [offers, setOffers] = useState([]);
  const [categories, setCategories] = useState([]);
  useEffect(() => {
    fetch("/api/banners")
      .then((res) => {
        return res.json();
      })
      .then((res) => {
        setOffers(res);
      })
      .catch((err) => {
        console.error(err);
      });
    fetch("/api/categories")
      .then((res) => {
        return res.json();
      })
      .then((res) => {
        setCategories(res);
      })
      .catch((err) => {
        console.error(err);
      });
  }, []);
  return (
    <React.Fragment>
      <Head
        title="Sabka Bazaar - Offers and Categories"
        description="Explore various categories of Products offered. Flat discounts on daily need and groceries "
      />
      <div className="home-container">
        {offers.length > 0 && (
          <Carousel items={offers.sort((a, b) => a.order - b.order)} />
        )}
        {categories.length > 0 &&
          categories
            .sort((a, b) => a.order - b.order)
            .map((cat) => <Card key={cat.order} item={cat} />)}
      </div>
    </React.Fragment>
  );
};

export default Home;
