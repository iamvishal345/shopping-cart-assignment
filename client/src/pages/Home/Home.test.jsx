import React from "react";
import { act } from "react-dom/test-utils";
import { mount } from "enzyme";
import Home from "./Home";
import fetchMock from "fetch-mock";
jest.mock("../../Utils/Head.jsx", () => ({
  Head: "div",
}));
jest.mock("../../components/Atoms/Carousel/Carousel.jsx", () => ({
  Carousel: "div",
}));
jest.mock("../../components/Atoms/Card/Card.jsx", () => ({
  Card: "div",
}));

const categories = [
  {
    name: "Beverages",
    key: "beverages",
    description:
      "Our beverage department will ensure your fridge is always fully stocked. Shop for soda, juice, beer and more. ",
    enabled: true,
    order: 3,
    imageUrl: "/static/images/category/beverages.png",
    id: "5b675e5e5936635728f9fc30",
  },
  {
    name: "Bakery Cakes and Dairy",
    key: "bakery-cakes-dairy",
    description:
      "The best cupcakes, cookies, cakes, pies, cheesecakes, fresh bread, biscotti, muffins, bagels, fresh coffee, milk and more.",
    enabled: true,
    order: 2,
    imageUrl: "/static/images/category/bakery.png",
    id: "5b6899123d1a866534f516de",
  },
  {
    name: "Beauty and Hygiene",
    key: "beauty-hygiene",
    description:
      "Buy beauty and personal care products online in India at best prices.",
    enabled: true,
    order: 4,
    imageUrl: "/static/images/category/beauty.png",
    id: "5b68994e3d1a866534f516df",
  },
];

const banners = [
  {
    bannerImageUrl: "/static/images/offers/offer1.jpg",
    bannerImageAlt: "Independence Day Deal - 25% off on shampoo",
    isActive: true,
    order: 1,
    id: "5b6c38156cb7d770b7010ccc",
  },
  {
    bannerImageUrl: "/static/images/offers/offer2.jpg",
    bannerImageAlt: "Independence Day Deal - Rs120 off on surf",
    isActive: true,
    order: 2,
    id: "5b6c38336cb7d770b7010ccd",
  },
  {
    bannerImageUrl: "/static/images/offers/offer3.jpg",
    bannerImageAlt: "Independence Day Deal - Rs99 off on domex",
    isActive: true,
    order: 3,
    id: "5b6c38456cb7d770b7010cce",
  },
  {
    bannerImageUrl: "/static/images/offers/offer4.jpg",
    bannerImageAlt: "Independence Day Deal - Rs99 off on bodywash",
    isActive: true,
    order: 4,
    id: "5b6c38576cb7d770b7010ccf",
  },
  {
    bannerImageUrl: "/static/images/offers/offer5.jpg",
    bannerImageAlt: "Independence Day Deal - Rs70 off on tea",
    isActive: true,
    order: 5,
    id: "5b6c386b6cb7d770b7010cd0",
  },
];

describe("Home", () => {
  it("should match the snapshot", () => {
    fetchMock.get("end:/api/banners", banners);
    fetchMock.get("end:/api/categories", categories);
    let wrapped;
    act(() => {
      wrapped = mount(<Home />);
    });
    expect(wrapped).toMatchSnapshot();
  });
});
