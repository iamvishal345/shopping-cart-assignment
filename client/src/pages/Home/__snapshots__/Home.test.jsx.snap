// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the snapshot 1`] = `
<Home>
  <div
    description="Explore various categories of Products offered. Flat discounts on daily need and groceries "
    title="Sabka Bazaar - Offers and Categories"
  />
  <div
    className="home-container"
  />
</Home>
`;
